// @clerk/nextjs@4.21.15/dist/esm/server/authenticateRequest.js
import { constants } from "@clerk/backend";
import { NextResponse } from "next/server";
import {
  API_KEY,
  CLERK_JS_URL,
  CLERK_JS_VERSION,
  clerkClient,
  debugRequestState,
  FRONTEND_API,
  PUBLISHABLE_KEY,
  SECRET_KEY
} from "./clerkClient";
import { apiEndpointUnauthorizedNextResponse, getCookie, handleMultiDomainAndProxy } from "./utils";
import { decorateResponseWithObservabilityHeaders } from "./withClerkMiddleware";
const authenticateRequest = async (req, opts) => {
  const { isSatellite, domain, signInUrl, proxyUrl } = handleMultiDomainAndProxy(req, opts);
  const cookieToken = getCookie(req, constants.Cookies.Session);
  const headers = req.headers;
  const headerToken = headers.get("authorization")?.replace("Bearer ", "");
  return await clerkClient.authenticateRequest({
    ...opts,
    apiKey: opts.apiKey || API_KEY,
    secretKey: opts.secretKey || SECRET_KEY,
    frontendApi: opts.frontendApi || FRONTEND_API,
    publishableKey: opts.publishableKey || PUBLISHABLE_KEY,
    isSatellite,
    domain,
    signInUrl,
    proxyUrl,
    cookieToken,
    headerToken,
    clientUat: getCookie(req, constants.Cookies.ClientUat),
    origin: headers.get("origin") || void 0,
    host: headers.get("host"),
    forwardedPort: headers.get("x-forwarded-port") || void 0,
    forwardedHost: headers.get("x-forwarded-host") || void 0,
    forwardedProto: headers.get("x-forwarded-proto") || void 0,
    referrer: headers.get("referer") || void 0,
    userAgent: headers.get("user-agent") || void 0,
    searchParams: new URL(req.url).searchParams
  });
};
const handleUnknownState = (requestState) => {
  const response = apiEndpointUnauthorizedNextResponse();
  decorateResponseWithObservabilityHeaders(response, requestState);
  return response;
};
const handleInterstitialState = (requestState, opts) => {
  const response = new NextResponse(
    clerkClient.localInterstitial({
      frontendApi: opts.frontendApi || FRONTEND_API,
      publishableKey: opts.publishableKey || PUBLISHABLE_KEY,
      clerkJSUrl: CLERK_JS_URL,
      clerkJSVersion: CLERK_JS_VERSION,
      proxyUrl: requestState.proxyUrl,
      isSatellite: requestState.isSatellite,
      domain: requestState.domain,
      debugData: debugRequestState(requestState),
      signInUrl: requestState.signInUrl
    }),
    {
      status: 401,
      headers: {
        "content-type": "text/html"
      }
    }
  );
  decorateResponseWithObservabilityHeaders(response, requestState);
  return response;
};
export {
  authenticateRequest,
  handleInterstitialState,
  handleUnknownState
};
